core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml

```diff
@@ -55,17 +55,31 @@
         --name epiphany-registry {{ specification.registry_image.name }}
       when: regitry_up_check.stdout | length == 0
 
+    - name: Set images for x86_64
+      set_fact:
+        generic_current_images: "{{ specification.images_to_load.x86_64.generic + specification.images_to_load.x86_64.current }}"
+        legacy_images: "{{ specification.images_to_load.x86_64.legacy }}"
+      when: ansible_architecture == "x86_64"
+
+    - name: Set images for aarch64
+      set_fact:
+        generic_current_images: "{{ specification.images_to_load.aarch64.generic + specification.images_to_load.aarch64.current }}"
+        legacy_images: "{{ specification.images_to_load.aarch64.legacy }}"
+      when: ansible_architecture == "aarch64"
+
     - name: Load generic and current version images
       vars:
         docker_image: "{{ item }}"
       include_tasks: load-image.yml
-      loop: "{{ specification.images_to_load.generic + specification.images_to_load.current }}"
+      loop: "{{ generic_current_images }}"
+      when: generic_current_images is iterable
```

Comment:
Here and in other similar places: Is this condition used to take into account not specified value? If not, what is the case to use such conditions for lists?

Reply:
For the default image definition for ARM inside ```/epiphany/core/src/epicli/data/common/defaults/configuration/image-registry.yml``` there is no legacy images. Even when defining the Yaml like below:

![image](https://user-images.githubusercontent.com/11056665/113709335-aa5e1580-96e2-11eb-8cd3-995993e25752.png)

It was giving some issues for the Ansible loops in the legacy images. So I added:

```yaml
when: generic_current_images is iterable
```
To fixe this. Added it to the other loops as well to avoid similar issues down the line.